### PR TITLE
EVG-6689: Add unconfigured patch status badge

### PR DESCRIPTION
**Pull request by SupaJoon**
[EVG-6689](jira.mongodb.com/browse/EVG-6689)

### Description
Screenshot 2023-05-23 at 2 30 33 PM

### CODE REVIEW

**Comment by ablack12** [comment]
Should we show unconfigured _instead of_ created? It kind of doesn't mean anything with that in mind. 

Re [our conventions](https://docs.google.com/document/d/1LJHPjLb_bUZxv-l_9Y61kitzAg7u-13I5DjOhZ-deBA/edit#heading=h.i9oftc5069w9) here we should have a verb in PR titles indicate what's actually happening (like "add unconfigured patch status badge")

Also, is it incidental that app is requested here rather than ui?

**Comment by ablack12** [comment]
The design that Minna shows in EVG-6689 does show just Unconfigured; I don't think this is a reasonable patch status for the backend though, and would be a massive change --can this be done on the front end without that? Adjusting the filter dropdown to created/unconfigured seems fine though. 

>Since non commit queue patches are shown alongside regular patches 
this sentence doesn't super make sense to me since non commit queue patches _are_ regular patches, is this a typo? Having mixed statuses is fine I think -- commit queue patches don't need configuring so showing created there would be correct.

File: src/components/PatchesPage/PatchCard.tsx [hunk 1]
```diff
@@ -106,7 +109,10 @@ export const PatchCard: React.VFC<Props> = ({
       </Left>
       <Center>
         <PatchBadgeContainer>
-          <PatchStatusBadge status={versionFull?.status ?? status} />
+          {!isUnconfigured && (
+            <PatchStatusBadge status={versionFull?.status ?? status} />
```
Comment (khelif96) [inline src/components/PatchesPage/PatchCard.tsx:113]: I think some change here broke the alignment
Before:
image

After:
image

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: src/components/PatchStatusBadge.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ export const PatchStatusBadge: React.VFC<Props> = ({ status }) => (
 );
 
 const statusToBadgeVariant = {
+  unconfigured: Variant.LightGray,
```
[the remark this answers is not in the document]
Comment (khelif96) [reply src/components/PatchStatusBadge.tsx:15]: Can you leave a comment here?

File: src/components/PatchesPage/PatchCard.tsx [hunk 3]
```diff
@@ -106,7 +109,10 @@ export const PatchCard: React.VFC<Props> = ({
...
+          )}
+          {isUnconfigured && <PatchStatusBadge status="unconfigured" />}
```
Comment (khelif96) [inline src/components/PatchesPage/PatchCard.tsx:115]:
```suggestion
          <PatchStatusBadge
            status={
              isUnconfigured ? "unconfigured" : versionFull?.status ?? status
            }
          />
```

File: src/utils/patch/index.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+interface IsPatchUnconfiguredParams {
+  alias: string;
+  activated: boolean;
+}
+
+export const isPatchUnconfigured = ({
+  alias,
+  activated,
+}: IsPatchUnconfiguredParams): boolean =>
+  !activated && alias !== commitQueueAlias;
```
Comment (khelif96) [inline src/utils/patch/index.ts:12]: nit. This function might read better as  
`isPatchConfigurable`.

File: src/components/PatchesPage/StatusSelector.tsx [hunk 5]
```diff
@@ -16,7 +16,9 @@ export const StatusSelector: React.VFC = () => {
     <Dropdown
       data-cy="my-patch-status-select"
       buttonText={`Patch Status: ${
-        statusVal.length ? statusVal.join(", ") : noFilterMessage
+        statusVal.length
+          ? statusVal.map((v) => statusValToCopy[v]).join(", ")
+          : noFilterMessage
```
Comment (khelif96) [inline src/components/PatchesPage/StatusSelector.tsx:21]: Nice!

**Review by khelif96 — APPROVED** [review]
Great job!

**Pull request merged**